extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/filter/OperationFilter.java

```diff
@@ -97,6 +100,20 @@ private String methodRef(Operation operation) {
         return (String) (extensions != null ? extensions.get(EXT_METHOD_REF) : null);
     }
 
+    private void maybeAddBadRequestResponse(Map.Entry<PathItem.HttpMethod, Operation> operation, String methodRef) {
+        if (!classNameMap.containsKey(methodRef)) {
+            return;
+        }
+        if (doAutoBadRequest && (operation.getKey().equals(PathItem.HttpMethod.POST)
+                || operation.getKey().equals(PathItem.HttpMethod.PUT))) { // Only applies to input
+            if (!operation.getValue().getResponses().hasAPIResponse("400")) { // Only when the user has not already added one
```

Comment:
This does not check if there is a body, just the verb. Is it the expected behavior?

Reply:
Don't you have to have a body with a PUT and a POST ?

Reply:
I am now checking for a requestbody

Reply:
So, no, any method can have a body (even if it's considered bad) and ... any method, including PUT and POST may not have a body...

Reply:
If a PUT and POST does not have a body, there will be no requestbody, so that is fine. (There are test cases for it)
I can not create a valid GET or DELETE endpoint that takes a requestbody. You get 415 (Invalid Media Type) always. This not a valid endpoint, and will always fail. I don't think we should auto document that. If a user want to create an endpoint that always returns a 415 on the GET, they can still add the OpenAPI annotation for that.